File: Python Fundamentals/Mid Exam problems/counter_strike.py
energy = int(input())

wins = 0

while True:
    command = input()

    if command == 'End of battle':
        print(f"Won battles: {wins}. Energy left: {energy}")
        break

    distance = int(command)

    if distance <= energy:
        energy -= distance
        wins += 1
    elif distance > energy:
        print(f"Not enough energy! Game ends with {wins} won battles and {energy} energy")
        break

    if wins % 3 == 0 and wins != 0:
        energy += wins

File: JavaScript Fundamentals/Mid Exams/Mid Exam 3/counterStrike.js
function solve(arr) {
    let energy = Number(arr.shift());
    let wins = 0;
    let lose = false;

    for (let el of arr) {
        if (el == "End of battle") break;
        if (wins % 3 == 0 && wins != 0) energy += wins;
        if (Number(el) <= energy) {
            energy -= Number(el);
            wins++;
        } else {
            lose = true;
            break;
        }
    }
    if (lose) {
        console.log(`Not enough energy! Game ends with ${wins} won battles and ${energy} energy`);
    } else {
        console.log(`Won battles: ${wins}. Energy left: ${energy}`);
    }
}
